### PR TITLE
Inventory groups frontend edits

### PR DESCRIPTION
**Pull request by perryr16**
#### What's this PR do?
- Adds db constraint to inventory groups requiring unique name
- Adds validation to group creation. The group and it's inventory must share an ALI.
- Adds ability to choose an ALI if an empty group is created
- Adds Group Map page
- Adds temp partials and controllers for inventory group dashboard, systems & services, and meters
- Various improvements

#### How should this be manually tested?
Sample test files
[ali_tree_better_buildings_lite.xlsx](https://github.com/user-attachments/files/17212873/ali_tree_better_buildings_lite.xlsx)
[AH_BB_250props100_lat_long.xlsx](https://github.com/user-attachments/files/17212843/AH_BB_250props100_lat_long.xlsx) 
[AH_BB_250props100_lat_long_diag.xlsx](https://github.com/user-attachments/files/17212842/AH_BB_250props100_lat_long_diag.xlsx) - Same properties, but properties lat has been updated

- Upload property info with ALI and lat/long info to cycle 1. Upload 2nd file to cycle 2.
- Go to  Inventory List > Groups > Create Group > create 2 groups for 2 different leaf nodes node
- Go to Inventory List > Select properties of a leaf node > Actions > Add to/Remove from Groups > Add button for the group with a matching ALI should be enabled, and the group with a mismatched ALI should be disabled
- Select properties from different ALIs > Actions > Add to/Remove from Groups > should see a warning that the selected views do not meet group requirements. Create is disabled, all adds are disabled.
- Go to Groups > click group name > navigate through Inventory Group Detail pages. The "Properties" and "Map" tab cycle dropdowns should update data to reflect cycle inventory

#### Screenshots (if appropriate)

Screenshot 2024-10-01 at 9 54 05 AM
Screenshot 2024-10-01 at 9 53 24 AM
Screenshot 2024-10-01 at 9 55 57 AM
Screenshot 2024-10-01 at 9 54 22 AM

### CODE REVIEW

**Review by haneslinger — COMMENTED** [review]
Just reviewed things server side.

File: seed/views/v3/access_levels.py [hunk 1]
```diff
@@ -491,3 +492,63 @@ def delete_instance(self, request, organization_pk=None, pk=None):
...
+        lowest_common = shared_ancestors.order_by("depth").last()
+
+        serialized_ali = AccessLevelInstanceSerializer(lowest_common).data
+        return JsonResponse({"status": "success", "data": serialized_ali})
+
+    @api_endpoint_class
+    @ajax_request_class
+    @has_perm_class("requires_viewer")
+    @action(detail=False, methods=["PUT"])
+    def filter_by_inventory(self, request, organization_pk=None):
```
[the remark this answers is not in the document]
Comment (perryr16) [reply seed/views/v3/access_levels.py:541]: Within /update_inventory_groups_modal_controller.js, this function is used to check the number of unique ALIs in a selection. If there is more than 1 unique ALI, the add to/create group buttons are disabled. If there is exactly 1, the ALI is used in the creation of the new group.

File: seed/utils/access_level_instance.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from django.core.exceptions import ObjectDoesNotExist
+
+from seed.models import AccessLevelInstance
+
+
+def access_level_filter(access_level_instance_id):
```
[the remark this answers is not in the document]
Comment (perryr16) [reply seed/utils/access_level_instance.py:6]: Could you elaborate on this? I was attempting to remove repetition when filtering for model instances within an ali. This pattern is used all over, and could be replaced by the filter in the above code, but I didn't want to make changes to dozens of files without approval.
```
self.model.objects.filter(
                    organization_id=self.get_organization(self.request),
                    property__access_level_instance__lft__gte=access_level_instance.lft,
                    property__access_level_instance__rgt__lte=access_level_instance.rgt,
                )
```

[the remark this answers is not in the document]
Comment (haneslinger) [reply seed/utils/access_level_instance.py:6]: firstly, the code you quoted above cannot be replaced by this function. it's missing "property__".

Secondly, if you _don't_ need the "property__", and are just looking for the tree under an ali, `.get_descendants` does that

[the remark this answers is not in the document]
Comment (perryr16) [reply seed/utils/access_level_instance.py:6]: You're right, the function would need to accept a prefix argument.

I played around with `.get_descendants()` and it's not quite right. `.get_descendants()` returns descendants of the node, but not the current node. So to get current and descendants, there are 4 db queries, while the original only uses 2. Let me know if you have a more elegant solution :)
```
access_level_instance = AccessLevelInstance.objects.get(pk=access_level_id)
current_and_descendants = AccessLevelInstance.objects.filter(id=access_level_id) | access_level_instance.get_descendants()
groups = groups.filter(access_level_instance__in=current_and_descendants)
```

[the remark this answers is not in the document]
Comment (haneslinger) [reply seed/utils/access_level_instance.py:6]: ah, you're right. Maybe `get_tree()`?

[the remark this answers is not in the document]
Comment (perryr16) [reply seed/utils/access_level_instance.py:6]: .get_tree returns the entire tree, sisters, aunts, ... everything

[the remark this answers is not in the document]
Comment (haneslinger) [reply seed/utils/access_level_instance.py:6]: even if you pass the current node as an arg? Its a class method

[the remark this answers is not in the document]
Comment (perryr16) [reply seed/utils/access_level_instance.py:6]: oh, thats perfect!

**Review by haneslinger — APPROVED** [review]

**Pull request merged**